src/body.rs

```diff
@@ -10,7 +13,7 @@ pub struct Body {
 
 impl Body {
     /// Instantiate a `Body` from a reader.
-    pub fn new<R: Read + 'static>(reader: R) -> Body {
+    pub fn new<R: Read + Seek + 'static>(reader: R) -> Body {
```

Comment:
I'm a bit ambivalent about that - I'd be a bit perplexed if two identically typed `Body` structs behaved completely differently when faced with a redirect.

It would be particularly annoying to deal with that in my case as cloudflare only intermittently returns the redirect depending on what it has cached - the prospect of trying to debug a failure that's intermittent in two dimensions (cloudflare, Body construction details) makes me nervous.

How about a `Body<Reusable>` and a `Body<Oneshot>`? I'd have to implement it to see what using it would look like, but it seems doable.